File: app/src/main/java/com/babestudios/companyinfouk/ui/search/SearchActivityView.java
package com.babestudios.companyinfouk.ui.search;

import com.babestudios.companyinfouk.data.model.search.CompanySearchResult;
import com.babestudios.companyinfouk.data.model.search.SearchHistoryItem;

import net.grandcentrix.thirtyinch.TiView;
import net.grandcentrix.thirtyinch.callonmainthread.CallOnMainThread;

import java.util.ArrayList;

interface SearchActivityView extends TiView {

	@CallOnMainThread
	void showProgress();

	@CallOnMainThread
	void hideProgress();

	@CallOnMainThread
	void showError();

	@CallOnMainThread
	void showRecentSearches(SearchHistoryItem[] searchHistoryItems);

	@CallOnMainThread
	void startCompanyActivity(String companyNumber, String companyName);

	@CallOnMainThread
	void showCompanySearchResult(CompanySearchResult companySearchResult, boolean isFromOnNext, SearchPresenter.FilterState filterState);

	@CallOnMainThread
	void clearSearchView();

	@CallOnMainThread
	void refreshRecentSearchesAdapter(ArrayList<SearchHistoryItem> searchHistoryItems);

	@CallOnMainThread
	void changeFabImage(SearchPresenter.FabImage type);

	@CallOnMainThread
	void showDeleteRecentSearchesDialog();

	void setFilterOnAdapter(SearchPresenter.FilterState filterState);

	void setDeferredFilterState(SearchPresenter.FilterState filterState);
}

File: app/src/main/java/com/babestudios/companyinfouk/ui/search/SearchActivityView.kt
package com.babestudios.companyinfouk.ui.search

import com.babestudios.companyinfouk.data.model.search.CompanySearchResult
import com.babestudios.companyinfouk.data.model.search.SearchHistoryItem

import net.grandcentrix.thirtyinch.TiView
import net.grandcentrix.thirtyinch.callonmainthread.CallOnMainThread

import java.util.ArrayList

interface SearchActivityView : TiView {

	@CallOnMainThread
	fun showProgress()

	@CallOnMainThread
	fun hideProgress()

	@CallOnMainThread
	fun showError()

	@CallOnMainThread
	fun showRecentSearches(searchHistoryItems: ArrayList<SearchHistoryItem>)

	@CallOnMainThread
	fun startCompanyActivity(companyNumber: String, companyName: String)

	@CallOnMainThread
	fun showCompanySearchResult(companySearchResult: CompanySearchResult, isFromOnNext: Boolean, filterState: SearchPresenter.FilterState)

	@CallOnMainThread
	fun clearSearchView()

	@CallOnMainThread
	fun refreshRecentSearchesAdapter(searchHistoryItems: ArrayList<SearchHistoryItem>)

	@CallOnMainThread
	fun changeFabImage(type: SearchPresenter.FabImage)

	@CallOnMainThread
	fun showDeleteRecentSearchesDialog()

	fun setFilterOnAdapter(filterState: SearchPresenter.FilterState)

	fun setInitialFilterState(filterState: SearchPresenter.FilterState)
}